### PR TITLE
MULE-16003: Investigate potentially unnecessary TCCL switches

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/policy/PolicyNextActionMessageProcessor.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
 import org.mule.runtime.core.api.MuleContext;
 import org.mule.runtime.core.api.context.notification.FlowStackElement;
 import org.mule.runtime.core.api.event.CoreEvent;
+import org.mule.runtime.core.api.processor.ContextClassloaderAwareProcessor;
 import org.mule.runtime.core.api.processor.Processor;
```
Comment (sonar-mulesoft) [inline core/src/main/java/org/mule/runtime/core/internal/policy/PolicyNextActionMessageProcessor.java]: Code Smell Code Smell: Remove this unused import 'org.mule.runtime.core.api.processor.Processor'. ([squid:UselessImportCheck](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7602&issues=AWkRT5ZPotIXnmsUSmGc&open=AWkRT5ZPotIXnmsUSmGc)

[ISSUE_KEY]: AWkRT5ZPotIXnmsUSmGc

File: core/src/main/java/org/mule/runtime/core/internal/util/rx/Operators.java [hunk 2]
```diff
@@ -40,13 +40,17 @@ private Operators() {}
...
+        if (event != null) {
+          CoreEvent result = mapper.apply(event);
+          if (result != null) {
+            sink.next(result);
+          } else {
+            ((BaseEventContext) event.getContext()).success();
+          }
         }
+      } catch (Exception e) {
+        sink.error(e);
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/util/rx/Operators.java]: @dfeist 
image

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Pull request closed without merging**